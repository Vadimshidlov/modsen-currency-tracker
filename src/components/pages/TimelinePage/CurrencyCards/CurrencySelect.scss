@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;

.currency-chart__form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  row-gap: 20px;



  .currency-chart__select-container {
    position: relative;
    display: flex;
    justify-content: center;
    row-gap: 20px;
    width: 30%;
    height: 50px;

    .currency-chart__select {
      /*outline: none;
      color: colors.$color_white;
      font-size: fonts.$font-size_30;
      width: 340px;
      padding: 5px;
      //font-size: 16px;
      line-height: 1;
      border: 0;
      border-radius: 5px;
      height: 34px;
      background: url('@/assets/svg/select-vector.svg') no-repeat right transparent;
      -webkit-appearance: none;
      background-position-x: 244px;*/
      width: 100%;
      border: none;
      outline: none;
      appearance: none;
      padding: 0 30px 0 15px;
      color: colors.$color_white;
      background-color: colors.$black_background;
      //background-color: #2873ff;
      @include fonts.PoppinsLight;
      font-size: fonts.$font-size_30;


      //.currency-chart-select__option {
      //  //background: transparent;
      //  margin: 40px;
      //  background: rgba(0, 0, 0, 0.3);
      //  color: #fff;
      //  text-shadow: 0 1px 0 rgba(0, 0, 0, 0.4);
      //}
    }

    .currency-chart__icon {
      width: 50px;
      height: 100%;
      position: absolute;
      right: 0;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
}

::-webkit-calendar-picker-indicator{
  background-color: colors.$color_white;
  padding: 5px;
  cursor: pointer;
  border-radius: 3px;
}
