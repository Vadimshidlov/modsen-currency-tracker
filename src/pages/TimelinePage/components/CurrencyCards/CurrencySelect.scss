@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;

.currency-chart__form {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;

    .currency-chart__select-container {
        position: relative;
        display: flex;
        justify-content: center;
        row-gap: 20px;
        width: 30%;
        height: 50px;

        @media (max-width: 1200px) {
            width: 260px;
        }

        .currency-chart__select {
            width: 100%;
            border: none;
            outline: none;
            appearance: none;
            padding: 0 30px 0 15px;
            color: colors.$color_white;
            background-color: colors.$black_background;
            @include fonts.PoppinsLight;
            font-size: fonts.$font-size_30;

            @media (max-width: 1200px) {
                font-size: fonts.$font-size_25;
            }

            @media (max-width: 1200px) {
                font-size: fonts.$font-size_20;
            }

            @media (max-width: 500px) {
                font-size: fonts.$font-size_15;
            }
        }

        .currency-chart__icon-container {
            display: flex;
            align-items: center;
            width: 20px;
            height: 100%;

            .currency-chart__icon {
                width: 20px;
                height: 15px;
                position: absolute;
                right: 0;
                display: flex;
                align-items: center;
                justify-content: center;

                @media (max-width: 1200px) {
                    width: 15px;
                    height: 15px;
                    font-size: fonts.$font-size_20;
                }
            }
        }
    }

    .currency-chart__date-picker-title {
        align-self: start;
        @include fonts.PoppinsLight;
        color: colors.$color_white;
        font-size: fonts.$font-size_15;
        margin-left: 20px;
        margin-bottom: 10px;

        @media (max-width: 500px) {
            margin-bottom: 0px;
            font-size: fonts.$font-size_12;
        }

        &.light {
            color: colors.$black_background;
        }
    }

    &.light {
        .currency-chart__select-container {
            .currency-chart__select {
                color: colors.$black_background;
                background-color: transparent;
            }
        }
    }
}

::-webkit-calendar-picker-indicator {
    background-color: colors.$color_white;
    padding: 5px;
    cursor: pointer;
    border-radius: 3px;
}
