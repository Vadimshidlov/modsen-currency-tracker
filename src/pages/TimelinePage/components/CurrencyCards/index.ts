export { default } from "@/pages/TimelinePage/components/CurrencyCards/CurrencyCards";
