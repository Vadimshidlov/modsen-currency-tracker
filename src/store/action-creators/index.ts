export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
