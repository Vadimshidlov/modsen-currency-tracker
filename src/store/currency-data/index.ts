export { mockMapData } from "@/store/currency-data/mockMapData";
export { mockBanksApiData } from "@/store/currency-data/mockBanksApiData";
export { mockStockData } from "@/store/currency-data/mockStockData";
