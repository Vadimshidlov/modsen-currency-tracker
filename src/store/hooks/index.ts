export { useActions } from "@/store/hooks/useActions";
export { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";
