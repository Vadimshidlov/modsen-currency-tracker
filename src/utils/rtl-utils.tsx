import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
