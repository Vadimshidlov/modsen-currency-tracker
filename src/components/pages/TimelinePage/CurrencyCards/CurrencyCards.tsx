import React, { Component, Dispatch } from "react";
import { connect } from "react-redux";
import {
    CurrentCurrencyActionType,
    IcurrentCurrencyState,
} from "@/store/reducers/latestCurrencyReducer";
import { getCurrency } from "@/store/action-creators/getCurrency";
import { RootStateType } from "@/store/reducers";
import "@/components/pages/TimelinePage/CurrencyCards/CurrencySelect.scss";
import { getCurrencyTitleWithCode } from "@/utils/getCurrencyWithCode";
import SelectIcon from "@/assets/svg/select-vector.svg";
// import { getRandomOhlcv } from "@/utils/getRandomOhlcv";
import ChartDate from "@/components/pages/TimelinePage/ChartDate/ChartDate";
import CurrencyCard from "@/components/pages/TimelinePage/CurrencyCard/CurrencyCard";
import { getSecondCurrency } from "@/utils/getSecondCurrency";
// import CurrencyCard from "@/components/pages/TimelinePage/CurrencyCard/CurrencyCard";

export type CurrencyCardsPropsType = {
    getCurrency(): (dispatch: Dispatch<CurrentCurrencyActionType>) => Promise<void>;
    currency: IcurrentCurrencyState;
};

export type CurrencyCardsStateType = {
    currencyCode: string;
};

class CurrencyCards extends Component<CurrencyCardsPropsType, CurrencyCardsStateType> {
    constructor(props: CurrencyCardsPropsType) {
        super(props);

        this.state = {
            currencyCode: "USD",
        };
        this.handleChangeCurrency = this.handleChangeCurrency.bind(this);
    }

    componentDidMount() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();

        // console.log(getRandomOhlcv(1));
    }

    componentDidUpdate() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    handleChangeCurrency(event: React.ChangeEvent<HTMLSelectElement>) {
        const { value } = event.target;

        this.setState({
            currencyCode: value,
        });
    }

    render() {
        const { currency } = this.props;
        const { currencyCode } = this.state;
        const { data } = currency.currency;
        const { secondCurrencyValue } = getSecondCurrency(currency.currency, currencyCode);

        return (
            <div className="currency-chart__form">
                <div className="currency-chart__select-container">
                    <select
                        className="currency-chart__select"
                        name="currency-chart-select"
                        id=""
                        value={currencyCode}
                        onChange={this.handleChangeCurrency}
                    >
                        {Object.keys(data).map((element) => (
                            <option
                                className="currency-chart-select__option"
                                value={element}
                                key={element}
                            >
                                {getCurrencyTitleWithCode(element)}
                            </option>
                        ))}
                    </select>
                    <div className="currency-chart__icon">
                        <SelectIcon width={20} height={15} />
                    </div>
                </div>
                <div className="currency-chart__date-from-container">
                    <ChartDate />
                </div>
                <div className="currency-chart__selected-card">
                    <CurrencyCard currencyCode={currencyCode} currencyValue={secondCurrencyValue} />
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state: RootStateType) => ({
    currency: state.currency,
});

const mapDispatchToProps = () => ({
    getCurrency,
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrencyCards);
