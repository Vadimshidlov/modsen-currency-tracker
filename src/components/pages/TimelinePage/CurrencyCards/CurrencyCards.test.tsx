import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import CurrencyCards from "@/components/pages/TimelinePage/CurrencyCards/CurrencyCards";

const mockResponse = {
    meta: {
        last_updated_at: "2024-03-04T23:59:59Z",
    },
    data: {
        USD: { code: "USD", value: 1 },
        EUR: { code: "EUR", value: 1 },
        BTC: { code: "BTC", value: 1 },
        JPY: { code: "JPY", value: 1 },
    },
};

describe("CurrencyCards component tests", () => {
    const onChange = jest.fn();

    it("should correct render component", async () => {
        renderWithProviders(
            <CurrencyCards
                currencyData={mockResponse}
                selectedCurrencyCode="USD"
                onChange={onChange}
            />,
        );

        screen.debug();

        expect(screen.getAllByText("Commercial Dollar")).toHaveLength(2);
        expect(screen.getByText("Euro")).toBeInTheDocument();
        expect(screen.getByText("Bitcoin")).toBeInTheDocument();
        expect(screen.getByText("Yen")).toBeInTheDocument();
        expect(screen.getByText("USD")).toBeInTheDocument();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(
            <CurrencyCards
                currencyData={mockResponse}
                selectedCurrencyCode="USD"
                onChange={onChange}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <CurrencyCards
                        currencyData={mockResponse}
                        selectedCurrencyCode="USD"
                        onChange={onChange}
                    />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
