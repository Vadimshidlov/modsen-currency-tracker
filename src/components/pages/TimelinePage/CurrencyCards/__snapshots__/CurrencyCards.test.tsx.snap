// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyCards component tests should match snapshot 1`] = `
[
  <div
    className="currency-chart__select-container"
  >
    <select
      className="currency-chart__select"
      name="currency-chart-select"
      onChange={[MockFunction]}
      value="USD"
    >
      <option
        className="currency-chart-select__option"
        value="USD"
      >
        Commercial Dollar
      </option>
      <option
        className="currency-chart-select__option"
        value="EUR"
      >
        Euro
      </option>
      <option
        className="currency-chart-select__option"
        value="BTC"
      >
        Bitcoin
      </option>
      <option
        className="currency-chart-select__option"
        value="JPY"
      >
        Yen
      </option>
    </select>
    <div
      className="currency-chart__icon-container"
    >
      <svg
        className="currency-chart__icon"
      />
    </div>
  </div>,
  <div
    className="chart-card__container"
  >
    <svg
      className="chart-card__icon"
    />
    <div
      className="chart-card__info"
    >
      <p
        className="chart-card__titile"
      >
        Commercial Dollar
      </p>
      <p
        className="chart-card__percent"
      >
        USD
      </p>
    </div>
  </div>,
]
`;

exports[`CurrencyCards component tests snapshot matches 1`] = `
<DocumentFragment>
  <div
    class="currency-chart__select-container"
  >
    <select
      class="currency-chart__select"
      name="currency-chart-select"
    >
      <option
        class="currency-chart-select__option"
        value="USD"
      >
        Commercial Dollar
      </option>
      <option
        class="currency-chart-select__option"
        value="EUR"
      >
        Euro
      </option>
      <option
        class="currency-chart-select__option"
        value="BTC"
      >
        Bitcoin
      </option>
      <option
        class="currency-chart-select__option"
        value="JPY"
      >
        Yen
      </option>
    </select>
    <div
      class="currency-chart__icon-container"
    >
      <svg
        class="currency-chart__icon"
      />
    </div>
  </div>
  <div
    class="chart-card__container"
  >
    <svg
      class="chart-card__icon"
    />
    <div
      class="chart-card__info"
    >
      <p
        class="chart-card__titile"
      >
        Commercial Dollar
      </p>
      <p
        class="chart-card__percent"
      >
        USD
      </p>
    </div>
  </div>
</DocumentFragment>
`;
